### DISCUSSION TITLE
Why does my Mermaid diagram render on the deployed GitHub pages site but not in the dev-server?

### DISCUSSION THREAD

**Discussion by kamorrissey** [squidfunk/mkdocs-material#5254] [Q&A] [answered]
I'm new to mkdocs and mkdocs-material, so let me know what diagnostic info would be helpful and where to get it.

I have a simple Mermaid.js graph diagram in a fenced code block that renders properly on my built and deployed GitHub Pages site, but when I use `mkdocs serve`, the diagram just renders as the Mermaid source. Why might this difference be?

I'm running the dev-server and dev browser inside an Ubuntu 20.04 VM from a Windows 10 host. Python 3.8.something. Don't know the mkdocs-material version, but I installed less than a week ago.

**Comment by squidfunk** [comment]
Hard to say without knowing exactly what you did. Try using a virtual environment. Other than that, the problem could be anywhere without knowing any more context. Check the browser console and see if there are any errors.

**Reply by kamorrissey** [reply]
I can't tell if the exception in the Firefox browser window has anything to do with the failure to render. The failure to render happens with both Firefox and Chromium in my VM. I haven't checked the Chromium console yet. I'm suspicious there is something going on in my VM. I am using a virtual python environment for mkdocs & mkdocs-material.

![Screenshot from 2023-03-24 18-13-05](https://user-images.githubusercontent.com/3477919/227659907-5544d94e-c0be-4ac1-8c3b-09e42654d711.png)

**Comment by kamorrissey** [comment, marked as answer]
I solved the problem by setting both browsers to autodetect the proxy settings. I need to see why my VM's proxy isn't set right.
